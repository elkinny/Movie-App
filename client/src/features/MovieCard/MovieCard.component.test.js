import MovieCardComponent from './MovieCard.component';
import React from 'react';
import { shallow } from 'enzyme';

describe('MovieCardComponent: ', () => {
  it('1. To match snapshot', () => {
    const props = {
      ref: () => {},
      duration: '120min',
      rating: '4.0',
      title: 'Captain Marvel',
      genre: 'Action',
      year: 2019,
      description:
        'Carol Danvers becomes one of the universe`s most powerful heroes when Earth is caught in the middle of a galactic war between two alien races.',
      link: 'https://www.youtube.com/watch?v=0LHxvxdRnYc',
      picLink:
        'https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SY1000_CR0,0,675,1000_AL_.jpg',
    };
    const component = shallow(<MovieCardComponent {...props} />);
    expect(component).toMatchSnapshot();
  });
});
