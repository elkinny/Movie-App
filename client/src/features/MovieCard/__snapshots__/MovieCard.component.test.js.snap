// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test:  MovieCardComponent to match snapshot 1`] = `
<Fragment>
  <div
    className="movie-card"
  >
    <div
      className="movie-card__header"
    >
      <img
        className="movie-card__img"
        height="400"
        src="https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SY1000_CR0,0,675,1000_AL_.jpg"
      />
    </div>
    <div
      className="movie-card__body"
    >
      <div
        className="movie-card__line"
      >
        <h2
          className="movie-card__title"
        >
          Captain Marvel
          <span
            className="movie-card__rating"
          >
             
            4.0
             
          </span>
        </h2>
      </div>
      <div
        className="movie-card__subline"
      >
        <span
          className="movie-card__genre"
        >
           
          Action
           
        </span>
        <span
          className="movie-card__year"
        >
           
          2019
           
        </span>
        <span
          className="movie-card__duration"
        >
           
          120min
           
        </span>
      </div>
      <p
        className="movie-card__description"
      >
        Carol Danvers becomes one of the universe\`s most powerful heroes when Earth is caught in the middle of a galactic war between two alien races.
      </p>
      <LinkButton
        className="movie-card__link-back"
        to="/"
      >
        Back
      </LinkButton>
    </div>
  </div>
</Fragment>
`;
